CMakeLists.txt

```
@@ -1,88 +1,58 @@
 
-if (WIN32)
-    if (NOT DEFINED ENV{JULIA_PATH})
-        message(FATAL_ERROR "Cannot determine location of julia image. Before running cmake, please manually set the environment variable JULIA_PATH using:\n\tset JULIA_PATH=\"C:/path/to/your/.../julia-1.7.2\"\nIf you are unsure of the location of your julia image, you can access the path from within the julia REPL using\n    println(joinpath(Sys.BINDIR, \"..\"))")
-    endif()
-else()
-    # access julia through julia environment variable
-    if (NOT DEFINED ENV{JULIA_PATH})
-        execute_process(
-                COMMAND julia -e "println(joinpath(Sys.BINDIR, \"..\"))"
-                OUTPUT_VARIABLE JULIA_PATH_LOCAL)
-
-        if ("${JULIA_PATH}" STREQUAL "")
-            message(FATAL_ERROR "Cannot determine location of julia image. Before running cmake, please manually set the environment variable JULIA_PATH using\n    export JULIA_PATH=/path/to/your/.../julia\nIf you are unsure of the location of your julia image, you can access the path from within the julia REPL using\n    println(joinpath(Sys.BINDIR, \"..\"))
-    For more information, visit https://github.com/Clemapfel/jluna/blob/master/README.md#troubleshooting")
-        endif()
-
-        set(ENV{JULIA_PATH} ${JULIA_PATH_LOCAL})
-    endif()
-endif()
+file(READ include/jluna.jl JULIA_SOURCE)
+set(C_ADAPTER_PATH ${CMAKE_INSTALL_PREFIX})
+configure_file("${CMAKE_SOURCE_DIR}/.src/include_julia.inl.in" "${CMAKE_SOURCE_DIR}/.src/include_julia.inl" @ONLY)
 
-if (WIN32)
-    set(JULIA_LIB "$ENV{JULIA_PATH}/lib/libjulia.dll.a")
-else()
-    set(JULIA_LIB "$ENV{JULIA_PATH}/lib/libjulia.so")
-endif()
-
-set(JULIA_INCLUDE "$ENV{JULIA_PATH}/include/julia")
-set(JULIA_FOUND TRUE)
-
-# verify shared library
-if (NOT EXISTS ${JULIA_LIB})
-    if (WIN32)
-        message(WARNING "Cannot find shared library libjulia.dll.a in $ENV{JULIA_PATH}/lib/")
-    else()
-        message(WARNING "Cannot find shared library libjulia.so in $ENV{JULIA_PATH}/lib/")
-    endif()
-    set(JULIA_FOUND FALSE)
-endif()
+### Declare C adapter ###
 
-# verify header
-if (NOT EXISTS ${JULIA_INCLUDE}/julia.h)
-    message(WARNING "Cannot find library header julia.h in $ENV{JULIA_PATH}/include/julia/")
-    set(JULIA_FOUND FALSE)
-endif()
-
-if (${JULIA_FOUND})
-    message("[LOG] Successfully detected julia image at $ENV{JULIA_PATH}")
-else()
-    message(FATAL_ERROR "Failed to detect julia image. Make sure JULIA_PATH is set correctly and that the julia image is uncompressed and not corrupted.\nFor more information, visit https://github.com/Clemapfel/jluna/blob/master/README.md#troubleshooting")
-endif()
-
-include_directories(${JULIA_INCLUDE})
-
-### JLUNA ###
+add_library(
+    jluna_c_adapter SHARED
+    .src/c_adapter.hpp
+    .src/c_adapter.cpp
+)
 
-include_directories(${CMAKE_SOURCE_DIR})
+target_compile_features(jluna_c_adapter PUBLIC cxx_std_20)
+target_include_directories(jluna_c_adapter PUBLIC "$<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}>")
```

You shouldn't assume that this project is always the top one and it doesn't really cost anything to not make such an assumption:
```suggestion
target_include_directories(jluna_c_adapter PUBLIC "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>")
```